src/plugins/data/common/search/aggs/buckets/histogram.ts

```diff
@@ -152,6 +156,29 @@ export const getHistogramBucketAgg = ({
             esTypes: aggConfig.params.field?.spec?.esTypes || [],
           });
         },
+        serialize(val, aggConfig) {
+          // store actually used auto interval in serialized agg config to be able to read it from the result data table meta information
+          const autoBounds = aggConfig?.getAutoBounds();
+          if (val === autoInterval && aggConfig && autoBounds) {
```

Comment:
This logic is duplicated with the `write` and `serialize` functions. I can't think of a better way, but maybe you can?